pkg/server/service.go

```diff
@@ -47,19 +49,25 @@ type rekorServer interface {
 type Server struct {
 	pb.UnimplementedRekorServer
 	grpc_health_v1.UnimplementedHealthServer
-	storage  tessera.Storage
-	readOnly bool
+	storage           tessera.Storage
+	readOnly          bool
+	algorithmRegistry *signature.AlgorithmRegistryConfig
 }
 
-func NewServer(storage tessera.Storage, readOnly bool) *Server {
+func NewServer(storage tessera.Storage, readOnly bool, allowedKeyAlgorithms []string) (*Server, error) {
 	if readOnly {
 		return &Server{
 			readOnly: readOnly,
-		}
+		}, nil
 	}
-	return &Server{
-		storage: storage,
+	algorithmRegistry, err := algorithmRegistry(allowedKeyAlgorithms)
```

Comment:
could we preparse this and avoid NewServer erroring? I don't know that that even matters, especially if changing this makes everything look worse.